### PR TITLE
PR 3 of 3: Mutation Signatures: Notebook

### PR DESCRIPTION
**Pull request by cansavvy**
#### Purpose/implementation

This third PR adds the main notebook of this analysis which calculates and plots mutation signatures for all samples using [COSMIC signatures](https://cancer.sanger.ac.uk/cosmic) and 
[Alexandrov et al, 2013](https://www.ncbi.nlm.nih.gov/pubmed/23945592) mutational signatures.

Note: This notebook currently downloads the SNV consensus file from S3 but this will not be how this is done after v10 is released. Also because of the manner of how the SNV consensus file is handled, it is not subset for CircleCI and won't be able to pass the CircleCI test until v10 is released and this is updated. 

#### Issue

This addresses #11 but it is related to and uses the data from #207 

#### Directions for reviewers

What are your thoughts about the results? 
Are there analyses you would like to see included that aren't? 

#### Results

Check out the rendered version of the notebook here: https://cansavvy.github.io/openpbta-notebook-concept/mutational-signatures/mutational_signatures.nb.html

#### Docker and continuous integration

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has not yet been added to continuous integration. (This will happen in the third PR).

#### PR Checklist
* [x] Run a linter
* [x] Set the seed (if applicable)
* [x] Comments and/or documentation up to date
* [x] Double check your paths
* [x] Check headline numbering
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

### CODE REVIEW

**Comment by jharenza** [comment]
@cansavvy I like the bubble plots. Minor visual - can you order the signatures numerically? Currently, they are alphabetical. Will make it easier to quickly assess signatures per histology.

**Comment by cansavvy** [comment]
Keeping in mind the v10 caveat, this is otherwise ready for an initial review, @jashapiro

File: analyses/mutational-signatures/mutational_signatures.Rmd [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+  dir.create(cosmic_plots, recursive = TRUE)
+}
+if (!dir.exists(nature_plots)) {
+  dir.create(nature_plots, recursive = TRUE)
+}
+```
+
+## Read in data
+
+Download and read in the consensus MAF file.
```
Comment (jashapiro) [inline analyses/mutational-signatures/mutational_signatures.Rmd]:
```suggestion
Read in the consensus MAF file. 
```

File: analyses/mutational-signatures/mutational_signatures.Rmd [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+```{r}
+# Declare file path for consensus file
+consensus_file <- file.path(data_dir, "snv-consensus_11122019", "consensus_mutation.maf.tsv")
+```
+
+Read in the consensus MAF file. 
+
+```{r}
+# Read in the file
+maf <- data.table::fread(consensus_file)
```
Comment (jashapiro) [inline analyses/mutational-signatures/mutational_signatures.Rmd]: Not sure this is needed, but...
```suggestion
maf <- data.table::fread(consensus_file, data.table = FALSE)
```

File: analyses/mutational-signatures/mutational_signatures.Rmd [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+```{r}
+ggplot2::ggsave(file.path(nature_plots, paste0("bubble_matrix_nature_mutation_sig.png")), width = 30, height = 20, units = "cm")
+```
+
+## Mutation signature grouped bar plots for each histology group
+
+We will make these plots for primary tumor samples only. 
+Let's make these for COSMIC mutation signatures first. 
+
+```{r}
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/mutational-signatures/mutational_signatures.Rmd]: Fair enough!

**Pull request merged**